typescript/lib/bulk_command_execution/v0_bulk_command_api/v0_bulk_commands.ts

```diff
@@ -0,0 +1,260 @@
+            return err(jsonErr);
+        }
+
+        this.deserializedCommandArgsPtr = args;
+        return ok(null);
+    }
+
+    public visitStartService(): Result<null, Error> {
+        let args: StartServiceArgs = newEmptyStartServiceArgs();
+
+        try {
+            args = JSON.parse(this.bytesToDeserialize);
+        } 
+        catch(jsonErr) {
+            return err(jsonErr);
+        }
+
+        this.deserializedCommandArgsPtr = args;
+        return ok(null);
+    }
+
+    public visitRemoveService(): Result<null, Error> {
+    	let args: RemoveServiceArgs = newEmptyRemoveServiceArgs();
+    	
+        try {
+            args = JSON.parse(this.bytesToDeserialize);
+        } 
+        catch(jsonErr) {
+            return err(jsonErr);
+        }
+
+        this.deserializedCommandArgsPtr = args;
+        return ok(null);
+    }
+
+    public visitRepartition(): Result<null, Error> {
+    	let args: RepartitionArgs = newEmptyRepartitionArgs();
+    	
+        try {
+            args = JSON.parse(this.bytesToDeserialize);
+        } 
+        catch(jsonErr) {
+            return err(jsonErr);
+        }
+
+        this.deserializedCommandArgsPtr = args;
+        return ok(null);
+    }
+
+    public visitExecCommand(): Result<null, Error> {
+        let args: ExecCommandArgs = newEmptyExecCommandArgs();
+     
+        try {
+            args = JSON.parse(this.bytesToDeserialize);
+        } 
+        catch(jsonErr) {
+            return err(jsonErr);
+        }
+
+        this.deserializedCommandArgsPtr = args;
+        return ok(null);
+    }
+
+    public visitWaitForEndpointAvailability(): Result<null, Error> {
+    	let args: WaitForEndpointAvailabilityArgs = newEmptyWaitForEndpointAvailabilityArgs();
+    	
+        try {
+            args = JSON.parse(this.bytesToDeserialize);
+        } 
+        catch(jsonErr) {
+            return err(jsonErr);
+        }
+
+        this.deserializedCommandArgsPtr = args;
+        return ok(null);
+    }
+
+    public visitExecuteBulkCommands(): Result<null, Error> {
+        let args: ExecuteBulkCommandsArgs = newEmptyExecuteBulkCommandsArgs();
+
+        try {
+            args = JSON.parse(this.bytesToDeserialize);
+        } 
+        catch(jsonErr) {
+            return err(jsonErr);
+        }
+
+        this.deserializedCommandArgsPtr = args;
+        return ok(null);
+    }
+
+    public getDeserializedCommandArgs(): jspb.Message {
+        return this.deserializedCommandArgsPtr;
+    }
+}
+
+// ====================================================================================================
+//                                        Serializable Command
+// ====================================================================================================
+
+class interstitialStruct {
+    private readonly type: V0CommandType;
+    private argsBytes: string; //TODO (comment) - original type was json.RawMessage ; couldn't find equivalent in typescript
```

Comment:
Ooo that `json.RawMessage` was a Golang feature I found. String wont' work here because the JSON message will look like:
`{
    "type":"someCommandType,
    "args":{
        "someProperty":"someValue",
        "someOtherProperty":"someOtherValue"
    }
}
```
String won't work because `args` is actually an object. In Go, setting the type to `json.RawMessage` was special - it meant, "capture all the bytes of the value" (in this case, of the "args" field) so that we can parse them later".